pkg/cloudfoundry/ManifestUtils.go

```diff
@@ -0,0 +1,226 @@
+	return toSlice(m.self)
+}
+
+// ApplicationHasProperty Checks if the application denoted by 'index' has the property 'name'
+func (m Manifest) ApplicationHasProperty(index int, name string) (bool, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return false, err
+	}
+
+	if index >= len(sliced) {
+		return false, fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	_m, err := toMap(sliced[index])
+
+	if err != nil {
+		return false, err
+	}
+
+	_, ok := _m[name]
+
+	return ok, nil
+}
+
+// GetApplicationProperty ...
+func (m Manifest) GetApplicationProperty(index int, name string) (interface{}, error) {
+
+	sliced, err := toSlice(m.self[propApplications])
+
+	if err != nil {
+		return "", err
+	}
+
+	if index >= len(sliced) {
+		return "", fmt.Errorf("Index (%d) out of bound. Number of apps: %d", index, len(sliced))
+	}
+
+	app, err := toMap(sliced[index])
+
+	if err != nil {
+		return "", err
+	}
+
+	if app[name] != nil {
+		return app[name], nil
+	}
+
+	return "", fmt.Errorf("No such property: '%s' available in application at position %d", name, index)
+}
+
+// GetAppName Gets the name of the app at 'index'
+func (m Manifest) GetAppName(index int) (string, error) {
+
+	appName, err := m.GetApplicationProperty(index, "name")
+
+	if err != nil {
+		return "", err
+	}
+
+	if name, ok := appName.(string); ok {
+		return name, nil
+	}
+
+	return "", fmt.Errorf("Cannot retrieve application name for app at index %d", index)
+}
+
+// Transform For each app in the manifest the first entry in the build packs list
+// gets moved to the top level under the key 'buildpack'. The 'buildpacks' list is
+// deleted.
+func (m *Manifest) Transform() error {
+
+	sliced, err := toSlice(m.self[propApplications])
+	if err != nil {
+		return err
+	}
+
+	for _, app := range sliced {
+		appAsMap, err := toMap(app)
+		if err != nil {
+			return err
+		}
+
+		err = transformApp(appAsMap, m)
+
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func transformApp(app map[string]interface{}, m *Manifest) error {
+
+	appName := "n/a"
+
+	if n, ok := app["name"].(string); ok {
+		if len(n) > 0 {
+			appName = n
+		}
+	}
+
+	if app[propBuildpacks] == nil {
+		// Revisit: not sure if a build pack is mandatory.
+		// In that case we should check that app.buildpack
+		// is present.
+		return nil
+	}
+
+	buildPacks, err := toSlice(app[propBuildpacks])
+
+	if err != nil {
+		return err
+	}
+
+	if len(buildPacks) > 1 {
+		return fmt.Errorf("More than one Cloud Foundry Buildpack is not supported. Please check manifest file '%s', application '%s'", m.name, appName)
+	}
+
+	if len(buildPacks) == 1 {
+		app[propBuildpack] = buildPacks[0]
+		delete(app, propBuildpacks)
+		m.modified = true
+	}
+
+	return nil
+}
+
+// HasModified ...
+func (m Manifest) HasModified() bool {
+	return m.modified
+}
+func toMap(i interface{}) (map[string]interface{}, error) {
```

Comment:
Why no empty line between these functions?

Reply:
done - for the sake of beautiness :-)